### PR TITLE
Phos refactor

### PR DESCRIPTION
**Pull request by ErinCain**
PHOS update draft

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -183,14 +236,28 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
                                fecundity = ..params$spawn_success_fecundity,
                                stochastic = stochastic)
 
-
-    # # For use in the r2r metrics
+    # R2R hatchery logic -------------------------------------------------------
+    # Currently adds only on major hatchery rivers (American, Battle, Feather, Merced, Moke)
+    # Add all as fry? or should we do larger
+    natural_juveniles <- rowSums(juveniles * natural_proportion_with_renat)
```
Comment (ergz) [inline R/model.R]: simple optimization: multiplying each col then adding is the same as adding them all then multiplying

```r
total_juves_pre_hatchery <- rowSums(juveniles)
natural_juveniles <- total_juves_pre_hatchery  * natural_proportion_with_renat
```

File: R/model.R [hunk 2]
```diff
@@ -658,49 +724,80 @@ fall_run_model <- function(scenario = NULL, mode = c("seed", "simulate", "calibr
...
+        rmultinom(1, (adults_in_ocean[i]), prob = c(.25, .5, .25))
       } else {
-        round(adults_in_ocean[i] * c(.25, .5, .25))
+        round((adults_in_ocean[i])* c(.25, .5, .25))
       }
-    }))
+    })) * output$proportion_natural_juves_in_tribs[ , year]
+
+    # TODO see if better place to do this
+    natural_adults_returning[is.na(natural_adults_returning)] = 0
```
Comment (ergz) [inline R/model.R]: i think you should switch to `is.nan`, NaN's are ok to appear in the code at this point (and be replaced) its just the places where denominator is 0, but if NA appears then something went wrong in the code you don't want to just replace those with zeros they should propagate and be fixed when they appear.

File: R/model.R (hunk 1 above)
Comment (ErinCain) [reply R/model.R]: updated!

File: R/model.R (hunk 2 above)
Comment (ErinCain) [reply R/model.R]: updated

**Comment by ErinCain** [comment]
Updated and removed draft status of PR - updates include: 
- updated hatchery release numbers 
- updated harvest rate 
- updated fecundity based on age (and associated model logic) 
- updated bookend params with new data objects ---TODO---still need the spawn decay multiplier in there. @ergz can you add this once you have rerun the spawn decay script in the R2R/DSMhabitat
- Initiated params for all the bookend scenarios __do not worry about reviewing these now because I will update all of them after the baseline calibration___

The results still seem reasonable but the new fecundity logic could definitely use a thorough review. I added some remaining TODOs in there for when we have time. It would be good to start functionalizing some of this R2R stuff where it it makes sense to clean up the script but I think that can wait until after this SDM call when we hopefully have a bit more time. 

Happy to hop on a quick call to discuss changes.

**Pull request merged**